### PR TITLE
Add integration_test page

### PR DESCRIPTION
**Pull request by johnpryan**
This adds instructions based on the [package:integration_test README](https://pub.dev/packages/integration_test) with additional instructions for running Android tests on Firebase TestLab

Staged: https://flutter-website-jr-stagi-988d4.firebaseapp.com/docs/testing/integration-tests/

### CODE REVIEW

**Comment by johnpryan** [comment]
CI is failing because the code snippets aren't set up correctly. Investigating...

File: src/docs/cookbook/testing/integration/index.md
Comment (sfshaza2) [inline src/docs/cookbook/testing/integration/index.md:6]: Indent note content by two spaces on each line. (By the way, Andrew is quite the expert on the code snippets now..). Also, code font `integration_test1` here and everywhere, except headers, alt text, etc.
Also, we use sentence case for titles, so it should be "Integration testing" with a lowercase t.

File: src/docs/cookbook/testing/integration/profiling.md
Comment (sfshaza2) [inline src/docs/cookbook/testing/integration/profiling.md:13]: Ditto

File: src/docs/cookbook/testing/integration/scrolling.md
Comment (sfshaza2) [inline src/docs/cookbook/testing/integration/scrolling.md:13]: Ditto

File: src/docs/perf/rendering/shader.md
Comment (sfshaza2) [inline src/docs/perf/rendering/shader.md:128]: Ditto

File: src/docs/testing/index.md
Comment (sfshaza2) [inline src/docs/testing/index.md:87]: I know this might seem like a real nit, but we recommend writing this sentence so that the reader can stop scanning if it doesn't apply. As it is now, you have to read the whole sentence. So:  "For information on how to write integration tests, see...."      https://developers.google.com/style/clause-order?hl=en

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:2]: Use a lowercase t.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:12]: Please be consistent about the empty brackets (which we prefer). So "[Firebase test lab][]"'.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:14]: Ditto

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:21]: Does this also work for desktop?

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:33]: Before you used "Firebase Test Lab". I googled and it looks like it is officially three words, not two.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:32]: =>>>.  .....on the target device, allowing you to run integration tests....       Also, these two short paragraphs beg some questions. Should they use both packages? Just integration_test? Are we phasing anything out? Is using just flutter_driver valid?  Also, code font the package names, here and elsewhere.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:36]: =>>> Project setup.  (sentence case).  https://developers.google.com/style/headings?hl=en

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:66]: Indent by 2 spaces.  Also, "or errors won't be..."

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:98]: sentence case

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:133]: Globally, use sentence case. I won't keep mentioning it.. Also, you MUST have at least one sentence between heading levels. So add a sentence here.  :)

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:144]: Add empty square brackets.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:145]: Menu items should be in bold, so `**Quality > Test Lab**`.   https://developers.google.com/style/ui-elements?hl=en

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:162]: You can add a link to that section.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:166]: Bold the menu item.   "This will start" ->> "This starts"  In fact, search for "will" in your doc. You can probably remove most of them.   https://developers.google.com/style/tense?hl=en

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:169]: Use sentence case for the alt text.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:171]: Bold these items. Only use title case if that is really what is in the UI.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:173]: Use asterisks for unordered lists, rather than minus signs.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:184]: Add empty square brackets.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:193]: Code font package names, etc.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:235]: Is the API in this code sample valid on stable now? If not, you need to preface this code with a skip tag. That could definitely explain the travis failure, though I didn't look at it.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:267]: and or -> or

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:10]: Use asterisk for unordered lists instead of minus signs.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:44]: Do you need a particular version of the sdk?

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:217]: Add a period to the end of the sentence. Be consistent. (Comment above ends with a period.)

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:220]: ditto

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:223]: Tab on the fab? What does this even mean? Is it emulating a tap? Maybe say that and end with a period.  :)

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:226]: Add period.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:253]: Period

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:256]: Ditto

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
Lots of little nits but, overall, I think it's good.. It does need a sentence or three about how the two packages coexist and when to use just one or both.

File: src/docs/testing/integration-tests/index.md
Comment (johnpryan) [reply src/docs/testing/integration-tests/index.md:169]: Isn't this already sentence case? Firebase Test Lab should be capitalized, right?

File: src/docs/testing/integration-tests/index.md
Comment (johnpryan) [reply src/docs/testing/integration-tests/index.md:237]: I just added sample code to the examples/ directory and prefixed these with the code-excerpt tag. so I think CI should pass now.

File: src/docs/testing/integration-tests/index.md
Comment (johnpryan) [reply src/docs/testing/integration-tests/index.md:44]: Yes, but that's enforced by `integration_test`'s pubspec, not by the user: https://github.com/flutter/plugins/blob/master/packages/integration_test/pubspec.yaml#L7

Comment (johnpryan) [reply src/docs/testing/integration-tests/index.md:44]: Actually, that's a Dart SDK constraint, so no it doesn't need a particular version I don't think. cc: @dnfield @nturgut

File: src/docs/testing/integration-tests/index.md
Comment (johnpryan) [inline src/docs/testing/integration-tests/index.md:42]: I added this paragraph to explain how flutter_driver, and integration_test relate to each other, and why integration_test is more familiar to those who have written a widget test before. LMWYT.

File: src/docs/testing/integration-tests/index.md
Comment (dnfield) [reply src/docs/testing/integration-tests/index.md:52]: This is pub's way of saying "just pull this from the local flutter SDK installation" rather than from pub.dev.

File: src/docs/testing/integration-tests/index.md
Comment (dnfield) [inline src/docs/testing/integration-tests/index.md:50]: This is soon going to change to 

```yaml
integration_test:
  sdk: flutter
```

Not sure if we should add a comment here to that effect, or at least a TODO for ourselves for it.

Comment (johnpryan) [reply src/docs/testing/integration-tests/index.md:50]: I created an issue to track for now: https://github.com/flutter/website/issues/5060 - is there an SDK issue I can follow?

Comment (johnpryan) [reply src/docs/testing/integration-tests/index.md:50]: It looks like it's already in the master channel, so I assume we're waiting for it to land in stable?

Comment (dnfield) [reply src/docs/testing/integration-tests/index.md:50]: Correct

File: src/docs/cookbook/testing/integration/index.md
Comment (sfshaza2) [inline src/docs/cookbook/testing/integration/index.md:7]: "Integration testing", global change.

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:74]: will not -> won't

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [reply src/docs/testing/integration-tests/index.md:177]: I'll grant you that.  :)

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:181]: What's this weird line?

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:206]: Code before migration:

File: src/docs/testing/integration-tests/index.md
Comment (sfshaza2) [inline src/docs/testing/integration-tests/index.md:243]: Put the "after" code into its own code block. Introduce with "Code after migration".

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
It looks like some of my feedback wasn't included yet?

**Review by nturgut — COMMENTED** [review]
Thanks for working on this? Who is the potential reader of this material? If it is web developers we also added a way for doing screenshot testing. It needs web driver. It is relatively flexible so the developers can attach their own comparator (for example we compare the image with the goldens repo)

Example usage: https://github.com/flutter/engine/blob/master/e2etests/web/regular_integration_tests/test_driver/text_editing_integration.dart#L47

README: https://github.com/flutter/engine/tree/master/e2etests/web/regular_integration_tests#adding-screenshot-tests

File: src/docs/testing/integration-tests/index.md
Comment (johnpryan) [reply src/docs/testing/integration-tests/index.md]: It already is in it's own block? I changed this to "Code after migration".

**Review by sfshaza2 — APPROVED** [review]
I think this is good enough to publish, though I may do some minor tweaks at a later time. LGTM. @johnpryan, I will let you merge this when ready. (I don't see that @dnfield or @nturgut have LGTM'd it.)

**Pull request merged**
